mirascope/anthropic/tools.py

```diff
@@ -109,20 +103,47 @@ def from_tool_call(cls, tool_call: ET.Element) -> AnthropicTool:
         Raises:
             ValidationError: if the tool call doesn't match the tool schema.
         """
-        try:
-            assert tool_call.tag == "invoke", "missing `<invoke>` root tag."
-        except AssertionError as e:
-            raise ValueError(f"Input `tool_call` XML had structure issues: {e}")
-
-        parameters: dict[str, Any] = {"tool_call": tool_call}
-        parameters_node = tool_call.find("parameters")
-        if parameters_node is not None:
-            parameters.update(
-                {
-                    parameter_node.tag: parameter_node.text
-                    for parameter_node in parameters_node
-                }
-            )
+
+        def _parse_xml_element(element: ET.Element) -> Union[str, dict, list]:
```

Comment:
is this intentionally a nested function?

Reply:
yeah, it's only used internally to this function and is private, and I thought it was easier to read. I pulled the other ones out because they got too long to be nested cleanly